src/main/java/net/kyori/adventure/text/minimessage/MiniMessage.java

```diff
@@ -221,6 +221,15 @@ static Builder builder() {
      */
     @NotNull Builder parsingErrorMessageConsumer(final Consumer<List<String>> consumer);
 
+    /**
+     * TODO write something here.
+     *
+     * @param compactingMethod method to use for compacting
+     * @return this builder
+     * @since 4.2.0
+     */
+    @NotNull Builder compactingMethod(Function<Component, Component> compactingMethod);
```

Comment:
I wonder if we should generify this even further as just a "post-serialization function thing" or something